tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestOperations/UnifiedInsertManyOperation.cs

```
@@ -0,0 +1,125 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format.UnifiedTestOperations
+{
+    public class UnifiedInsertManyOperation : IUnifiedTestOperation
+    {
+        private IMongoCollection<BsonDocument> _collection;
+        private List<BsonDocument> _documents;
+        private InsertManyOptions _options;
+        private IClientSessionHandle _session;
+
+        public UnifiedInsertManyOperation(
+            IMongoCollection<BsonDocument> collection,
+            List<BsonDocument> documents,
+            InsertManyOptions options,
+            IClientSessionHandle session)
+        {
+            _collection = collection;
+            _documents = documents;
+            _options = options; // TODO: should not be null. Either throw or recreate
+            _session = session;
+        }
+
+        public OperationResult Execute(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (_session == null)
+                {
+                    _collection.InsertMany(_documents, _options, cancellationToken);
+                }
+                else
+                {
+                    _collection.InsertMany(_session, _documents, _options, cancellationToken);
+                }
```

I would return here instead of on line 62:

    return null;

---

Is `null` really the correct return value? Should we instead return:

    return new OperationResult((BsonValue)null);

---

> Is null really the correct return value?

My bad, missed it. This implementation is quite dangerous in terms of these kind of errors, since the assert will just be silently skipped when result set to `null`.

> I would return here instead of on line 62

I really like the fact that we don't have to write the line twice. Should I still go for the change?

---

No need to write the return line twice.

But move it to after line 55.

---

Done

---

This is what I'm suggesting:

    public OperationResult Execute(CancellationToken cancellationToken)
    {
        try
        {
            if (_session == null)
            {
                _collection.InsertMany(_documents, _options, cancellationToken);
            }
            else
            {
                _collection.InsertMany(_session, _documents, _options, cancellationToken);
            }
            return new OperationResult((BsonValue)null);
        }
        catch (Exception exception)
        {
            return new OperationResult(exception);
        }
    }